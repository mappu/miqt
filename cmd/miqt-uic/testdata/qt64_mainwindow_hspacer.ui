<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>785</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
