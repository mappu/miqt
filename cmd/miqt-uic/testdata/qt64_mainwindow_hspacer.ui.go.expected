// Generated by miqt-uic. To update this file, edit the .ui file in
// Qt Designer, and then run 'go generate'.
//
//go:generate miqt-uic go_generate_args

package test_package_name

import (
	qt "github.com/mappu/miqt/qt6"
)

type MainWindowUi struct {
	MainWindow       *qt.QMainWindow
	centralwidget    *qt.QWidget
	verticalLayout   *qt.QVBoxLayout
	horizontalSpacer *qt.QSpacerItem
	menubar          *qt.QMenuBar
	statusbar        *qt.QStatusBar
}

// NewMainWindowUi creates all Qt widget classes for MainWindow.
func NewMainWindowUi() *MainWindowUi {
	ui := &MainWindowUi{}
	ui.MainWindow = qt.NewQMainWindow(nil)
	MainWindow__objectName := qt.NewQAnyStringView3("MainWindow")
	ui.MainWindow.SetObjectName(*MainWindow__objectName)
	MainWindow__objectName.Delete() // setter copied value
	ui.MainWindow.Resize(800, 600)
	ui.centralwidget = qt.NewQWidget(ui.MainWindow.QWidget)
	centralwidget__objectName := qt.NewQAnyStringView3("centralwidget")
	ui.centralwidget.SetObjectName(*centralwidget__objectName)
	centralwidget__objectName.Delete() // setter copied value
	ui.verticalLayout = qt.NewQVBoxLayout(ui.centralwidget)
	verticalLayout__objectName := qt.NewQAnyStringView3("verticalLayout")
	ui.verticalLayout.SetObjectName(*verticalLayout__objectName)
	verticalLayout__objectName.Delete() // setter copied value
	ui.verticalLayout.SetContentsMargins(11, 11, 11, 11)
	ui.verticalLayout.SetSpacing(6)
	ui.horizontalSpacer = qt.NewQSpacerItem4(785, 20, qt.QSizePolicy__Expanding, qt.QSizePolicy__Minimum)

	ui.verticalLayout.AddItem(ui.horizontalSpacer.QLayoutItem)
	ui.MainWindow.SetCentralWidget(ui.centralwidget) // Set central widget
	ui.menubar = qt.NewQMenuBar(ui.MainWindow.QWidget)
	menubar__objectName := qt.NewQAnyStringView3("menubar")
	ui.menubar.SetObjectName(*menubar__objectName)
	menubar__objectName.Delete() // setter copied value
	ui.menubar.Resize(800, 22)
	ui.MainWindow.SetMenuBar(ui.menubar)
	ui.statusbar = qt.NewQStatusBar(ui.MainWindow.QWidget)
	statusbar__objectName := qt.NewQAnyStringView3("statusbar")
	ui.statusbar.SetObjectName(*statusbar__objectName)
	statusbar__objectName.Delete() // setter copied value
	ui.MainWindow.SetStatusBar(ui.statusbar)

	ui.Retranslate()

	return ui
}

// Retranslate reapplies all text translations.
func (ui *MainWindowUi) Retranslate() {
	ui.MainWindow.SetWindowTitle(qt.QCoreApplication_Tr("MainWindow"))
}
